server/leafnode.go

```diff
@@ -2269,6 +2279,13 @@ func (c *client) leafPermViolation(pub bool, subj []byte) {
 
 // Invoked from generic processErr() for LEAF connections.
 func (c *client) leafProcessErr(errStr string) {
+	// Check if we got a cluster name collision.
+	if strings.Contains(errStr, "same cluster name") {
```

Comment:
I wonder if we should not be more strict here and use `ErrLeafNodeHasSameClusterName.Error()` so that we don't have another error some day that contains this string but is not really the error we are dealing with here?

Reply:
I can make that change, sounds good.